### ISSUE TITLE
Lots of "bad udp cksum" when talking to LiteEth

### ISSUE THREAD

**Issue by mithro** [enjoy-digital/opsis-soc#30]
```
00:04:27.025295 IP (tos 0x0, ttl 64, id 0, offset 0, flags [DF], proto UDP (17), length 32)
    192.168.1.50.7642 > 192.168.1.100.55970: [udp sum ok] UDP, length 4
00:04:27.042807 IP (tos 0x0, ttl 64, id 0, offset 0, flags [DF], proto UDP (17), length 48)
    192.168.1.50.7642 > 192.168.1.100.69: [udp sum ok]  20 RRQ "cmdline.txt" octet
00:04:27.043409 IP (tos 0x0, ttl 64, id 2649, offset 0, flags [DF], proto UDP (17), length 47)
    192.168.1.100.60173 > 192.168.1.50.7642: [bad udp cksum 0x8413 -> 0x31f2!] UDP, length 19
00:04:27.073294 IP (tos 0x0, ttl 64, id 0, offset 0, flags [DF], proto UDP (17), length 47)
    192.168.1.50.7642 > 192.168.1.100.69: [udp sum ok]  19 RRQ "initrd.bin" octet 
00:04:27.073858 IP (tos 0x0, ttl 64, id 2653, offset 0, flags [DF], proto UDP (17), length 47)
    192.168.1.100.37725 > 192.168.1.50.7642: [bad udp cksum 0x8413 -> 0x89a2!] UDP, length 19
00:04:27.349557 IP (tos 0x0, ttl 1, id 48751, offset 0, flags [DF], proto UDP (17), length 200)
    192.168.1.100.57978 > 239.255.255.250.1900: [bad udp cksum 0xb2cc -> 0xc65a!] UDP, length 172
00:04:27.365784 IP (tos 0x0, ttl 1, id 48753, offset 0, flags [DF], proto UDP (17), length 200)
    192.168.1.100.39783 > 239.255.255.250.1900: [bad udp cksum 0xb2cc -> 0x0d6e!] UDP, length 172
00:04:28.349953 IP (tos 0x0, ttl 1, id 48925, offset 0, flags [DF], proto UDP (17), length 200)
    192.168.1.100.57978 > 239.255.255.250.1900: [bad udp cksum 0xb2cc -> 0xc65a!] UDP, length 172
00:04:28.366559 IP (tos 0x0, ttl 1, id 48926, offset 0, flags [DF], proto UDP (17), length 200)
    192.168.1.100.39783 > 239.255.255.250.1900: [bad udp cksum 0xb2cc -> 0x0d6e!] UDP, length 172
00:04:29.350492 IP (tos 0x0, ttl 1, id 49130, offset 0, flags [DF], proto UDP (17), length 200)
    192.168.1.100.57978 > 239.255.255.250.1900: [bad udp cksum 0xb2cc -> 0xc65a!] UDP, length 172
00:04:29.367481 IP (tos 0x0, ttl 1, id 49133, offset 0, flags [DF], proto UDP (17), length 200)
    192.168.1.100.39783 > 239.255.255.250.1900: [bad udp cksum 0xb2cc -> 0x0d6e!] UDP, length 172
00:04:30.351568 IP (tos 0x0, ttl 1, id 49173, offset 0, flags [DF], proto UDP (17), length 200)
    192.168.1.100.57978 > 239.255.255.250.1900: [bad udp cksum 0xb2cc -> 0xc65a!] UDP, length 172
00:04:30.367616 IP (tos 0x0, ttl 1, id 49175, offset 0, flags [DF], proto UDP (17), length 200)
```

**Comment by enjoy-digital** [comment]
Was it captured with the opsis_minisoc or opsis_video target?

**Comment by mithro** [comment]
This was on the opsis_minisoc target. The tftp boot worked, but tcpdump showed a lot of packets with bad checksums. I can get a more detailed capture if you want.

**Comment by mithro** [comment]
Sorry, let me try again...

On the opsis_minisoc target, the tftp boot worked. In tcpdump you could see a bunch of packets with good checksums and a bunch of packets with bad checksums.

The opsis_video target, the tftp boot doesn't work. In tcpdump you see a bunch of packets with bad checksums.

I can get a more detailed captures if you want.

**Comment by enjoy-digital** [comment]
OK that's interesting since it shows that there is something already wrong on the opsis_minisoc target.
The issue is in the RGMII PHY: https://github.com/enjoy-digital/liteeth/blob/master/liteeth/phy/s6rgmii.py
Just need to figure out where or if it's a clock domain that is not constrainted correctly.

**Comment by enjoy-digital** [comment]
Fixed via https://github.com/enjoy-digital/opsis-soc/commit/4aee8f102d668967f859d26052c1963c0295e643.
